src/test_validator.py

```diff
@@ -0,0 +1,180 @@
+import unittest
+import datetime as dt
+from validation import String, Date
+
+def broken_function():
+    raise Exception('This is broken')
+
+class TestValidator(unittest.TestCase):
+    
+    #default case is not handled in the code
+    def test_string(self):
+        test_str = String(case="lowr")
+        #assert test_str.validate('kg') != None
+        
+    def test_string_lower(self):
+        
+        def test_string_lower_wrong(data):
+            flag = False
+            test_str = String(case="lower")
+            try:
+                test_str.validate(data)
+            except Exception as e:
```

Comment:
Specifying the expected error rather than using Exception will ensure we catch what we what. I'd say the best option, rather than using this flag system would be to import pytest and use something like the following:
``` python
with pytest.raises(TypeError):
    test_str.validate(data)
```
Doing this will save you a lot of lines of code needing to be written, as it passes the assertion if the error is raised. This also means you won't need to use a flag system and can instead just write the statement that should raise the error.